app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,70 @@
+class GroupsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_group, only: %i[show edit update destroy]
+
+  # GET /groups or /groups.json
+  def index
+    @groups = Group.includes([groups_entities: :entity]).where(author_id: current_user.id)
+  end
+
+  # GET /groups/1 or /groups/1.json
+  def show; end
+
+  # GET /groups/new
+  def new
+    @group = Group.new
+  end
+
+  # GET /groups/1/edit
+  def edit; end
+
+  # POST /groups or /groups.json
+  def create
+    @group = Group.new(group_params)
+
+    respond_to do |format|
+      if @group.save
+        format.html { redirect_to groups_path, notice: 'Category was successfully created.' }
+        format.json { render :show, status: :created, location: @group }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @group.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /groups/1 or /groups/1.json
+  def update
+    respond_to do |format|
+      if @group.update(group_params)
+        format.html { redirect_to group_url(@group), notice: 'Category was successfully updated.' }
+        format.json { render :show, status: :ok, location: @group }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @group.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /groups/1 or /groups/1.json
+  def destroy
+    @group.destroy
+
+    respond_to do |format|
+      format.html { redirect_to groups_url, notice: 'Category was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_group
+    @group = Group.find(params[:id])
+  end
+
+  # Only allow a list of trusted parameters through.
+  def group_params
+    params.require(:group).permit(:name, :icon, :author_id)
```

Comment:
Please ensure that you add `image` to the category as in the design instead of using text. You can use `active storage` to save images to your database.

Reply:
Hi @micheaol, thanks for your kind feedback. In this one, I might differ because I cannot find specified in any place of the [Microverse Documentation](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md) that it has to be an image. Apart from that, I already consult with other partners and no one was asked to do this. And the worst-case scenario is that the implementation of what you asked for (active storage) is something that we didn't cover in any lesson in the past weeks. So please, reconsider your thoughts about this one.

Reply:
@TheLuisAbarca ,

You do have to use active storage, I just suggested that you can. But the project requirement says, your app should look like the design and adding images, not text will make it look the same.

Reply:
@TheLuisAbarca you are correct as long as you can upload them using URL but consider this as an OPTIONAL change and try to learn more about how to upload storage using active storage as a practice for you despite it is required by Microverse or not, I can help you with it if you got stuck.


[How to upload Image in Rails using active storage](https://medium.com/swlh/how-to-upload-images-into-your-rails-project-using-active-storage-1285a69e4cf5)